src/server.rs

```diff
@@ -373,10 +373,17 @@ pub extern "C" fn rustls_server_session_process_new_packets(
 ) -> rustls_result {
     ffi_panic_boundary! {
         let session: &mut Sess = try_mut_from_ptr!(session);
-        let _guard = userdata_push(session.userdata);
-        match session.session.process_new_packets() {
+        let guard = match userdata_push(session.userdata) {
```

Comment:
The `and_then()` combinator would make this a bit less repetitive, I think.

Reply:
I couldn't quite figure out an elegant way to do this with `and_then()` but if you'd like to show an example of the direction you're thinking in, I'm happy to try it. Thanks for all the tips on combinators, they are nice improvements.